sklearn/model_selection/_validation.py

```diff
@@ -400,8 +400,13 @@ def cross_validate(
             raise UnsetMetadataPassedError(
                 message=(
                     f"{sorted(e.unrequested_params.keys())} are passed to cross"
-                    " validation but are not explicitly requested or unrequested. See"
-                    " the Metadata Routing User guide"
+                    " validation but are not explicitly set as requested or not"
+                    " requested for cross_validate's estimator:"
+                    f" {estimator.__class__.__name__}. Call"
+                    " `.set_fit_request({{metadata}}=True)` on the estimator for"
+                    f" each metadata in {sorted(e.unrequested_params.keys())} that you"
```

Comment:
Nit: Since `sorted(e.unrequested_params)` is used twice, can you set it as a variable and use it in the message?

```python
unrequested_params = sorted(e.unrequested_params)

message = (
    f"{unrequested_params}"
    ...
```

Also `sorted(dictionary)` will iterate over the dict's keys, so `keys` is not required. I have seen some prefer the `keys`, so I am okay with keeping it if you think it is easier to understand.